Sources/Build/TestMetadataParser.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Utility
+import PackageType
+
+struct TestClassMetadata {
+    let name: String
+    let testNames: [String]
+}
+
+struct ModuleTestMetadata {
+    let module: TestModule
+    let testManifestPath: String
+    let dependencies: [String]
+    let classes: [TestClassMetadata]
+}
+
+protocol TestMetadataParser {
+    func parseTestClasses(testFilePath: String) throws -> [TestClassMetadata]?
+}
+
+//up for grabs
+//struct ASTTestMetadataParser: TestMetadataParser {
+//    
+//
+//}
+
+struct StringTestMetadataParser: TestMetadataParser {
+    
+    /// Based on the absolute path passed in parses test class metadata
+    /// Returns array because there might be more test classes in one file
+    func parseTestClasses(testFilePath: String) throws -> [TestClassMetadata]? {
+        
+        guard let lines = try? File(path: testFilePath).enumerate() else { return nil }
+        
+        var classes = [TestClassMetadata]()
+        func finishClass(className: String?, _ testNames: [String]) {
+            if let className = className where testNames.count > 0 {
+                let testClass = TestClassMetadata(name: className, testNames: testNames)
+                classes.append(testClass)
+            }
+        }
+        
+        //parse lines one by one, no need to load the whole into memory
+        var className: String? = nil
+        var testNames = [String]()
+        while true {
+            
+            guard let line = lines.next() else {
+                //EOF, no more classes
+                finishClass(className, testNames)
+                testNames = []
+                return classes.nilIfEmpty()
+            }
+            
+            //look for a class name
+            if let newClassName = parseClassName(line) {
+                //found class name, finish old class and start new
+                finishClass(className, testNames)
+                className = newClassName
+                testNames = []
+            }
+            
+            //look for a test case
+            if let testName = parseTestName(line) {
+                //found another test name
+                testNames.append(testName)
+            }
+        }
+    }
+    
+    func parseClassName(line: String) -> String? {
+        var comps = line.splitWithCharactersInString(":\r\t\n ").filter { !$0.isEmpty }
+        guard comps.count >= 2 else { return nil }
+        //see if this line is a candidate
+        guard Set(comps).contains("class") else { return nil }
+        while true {
+            let allowedFirst = Set(["private", "public", "internal", "final"])
```

Comment:
As things are now, we will still generate a manifest, but the compiler will fail on not being able to reference `MySecretTestCase` from its module's corresponding `LinuxTestManifest.swift`. I wonder what we *should* do here, I think erroring out is fine. Do private test classes even work?

TBH, hopefully this would become easier if [SEP-0025](https://github.com/apple/swift-evolution/blob/master/proposals/0025-scoped-access-level.md) passed, but for the time being, I'm accepting all ideas.

Obviously, we could detect `private` here right away and throw a very specific error that would tell the use to make their test class at least `internal`, but since this is a parser, I'd prefer it to not have any side effects (but could live with it).